File: clojure/scrabble-score/src/scrabble_score.clj
(ns scrabble-score
  (:require [clojure.string :as str]))

(defn score-letter [c]
  (case (str (str/lower-case c))
    ("a" "e" "i" "o" "u" "l" "n" "r" "s" "t") 1
    ("d" "g") 2
    ("b" "c" "m" "p") 3
    ("f" "h" "v" "w" "y") 4
    "k" 5
    ("j" "x") 8
    ("q" "z") 10))

(defn score-word [s]
  (transduce (map score-letter) + 0 s))

File: ruby/scrabble-score/scrabble_score.rb
class Scrabble
  def initialize(word)
    @word = word
  end

  def score_for_letter(letter)
    case letter
    when "A", "E", "I", "O", "U", "L", "N", "R", "S", "T" then 1
    when "D", "G" then 2
    when "B", "C", "M", "P" then 3
    when "F", "H", "V", "W", "Y" then 4
    when "K" then 5
    when "J", "X" then 8
    when "Q", "Z" then 10
    end
  end

  def score
    @word
      .upcase
      .chars
      .sum { |l| score_for_letter(l) }
  end
end
